confmap/provider_test.go

```diff
@@ -6,13 +6,87 @@ package confmap
 import (
 	"context"
 	"errors"
+	"fmt"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+// This is an example of a provider that calls a provided WatcherFunc to update configuration dynamically every second.
+// The example is useful for implementing Providers of configuration that changes over time.
+type UpdatingProvider struct{}
+
+func (p UpdatingProvider) getCurrentConfig(_ string) any {
+	return "hello"
+}
+
+func (p UpdatingProvider) Retrieve(ctx context.Context, uri string, watcher WatcherFunc) (*Retrieved, error) {
+	ticker := time.NewTicker(1 * time.Second)
+	stop := make(chan bool, 1)
+
+	retrieved, err := NewRetrieved(p.getCurrentConfig(uri), WithRetrievedClose(func(_ context.Context) error {
+		// the provider should call this function when it no longer wants config updates
+		ticker.Stop()
+		stop <- true
+		return nil
+	}))
+	if err != nil {
+		return nil, err
+	}
+	// it's necessary to start a go function that can notify the caller of changes asynchronously
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				// if the context is closed, then we should stop sending updates
+				ticker.Stop()
+				return
+			case <-stop:
+				// closeFunc was called, so stop updating the watcher
+				fmt.Println("unsubscribing watcher")
```

Comment:
```suggestion
				t.Println("unsubscribing watcher")
```
Could we put a `*testing.T` on the `UpdatingProvider` type and use that to print here?